lib/class-wp-theme-json-gutenberg.php

```diff
@@ -1049,6 +1049,23 @@ private function get_block_classes( $style_nodes ) {
 				}
 			}
 
+			// Set a default margin on the root selector before generating the ruleset
+			// from the `theme.json`. This is to ensure that if the `theme.json` declares
+			// `margin` in its `spacing` declaration for the `body` element then these
+			// user-generated values take precedence in the CSS cascade.
+			// See: https://github.com/WordPress/gutenberg/issues/36147.
+			if ( self::ROOT_BLOCK_SELECTOR === $selector ) {
```

Comment:
By applying the margin: 0 rule early we ensure that any values provided from `theme.json` will be later in the selector order and thus take precedence over the default.